clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1491,7 +1490,7 @@ public void assign(Collection<TopicPartition> partitions) {
             // See the ApplicationEventProcessor.process() method that handles this event for more detail.
             applicationEventHandler.add(new AssignmentChangeEvent(subscriptions.allConsumed(), time.milliseconds()));
 
-            log.info("Assigned to partition(s): {}", join(partitions, ", "));
+            log.info("Assigned to partition(s): {}", String.join(", ", Arrays.toString(partitions.toArray())));
```

Comment:
`Arrays.toString` will add `[]` so that is a bit different to origin message. Maybe you can use the lambda to generate same message.

Reply:
I'm thinking to append `replace("[", "").replace("]", "")` to eliminate brackets. If these's any better representations please let me know. Thanks!

Reply:
how about `partitions.stream().map(TopicPartition::toString).collect(Collectors.joining(","))`?

Reply:
Got it, thanks for the advice!